backend/handlers.go

```diff
@@ -49,7 +59,33 @@ func (cs *server) publishHandler(w http.ResponseWriter, r *http.Request, _ httpr
 	}
 	defer body.Close()
 
-	cs.publish(msg)
+	// Unmarshal the JSON message into a map
+	var data map[string]interface{}
+	if err := json.Unmarshal(msg, &data); err != nil {
+		http.Error(w, "Invalid JSON payload", http.StatusBadRequest)
+		return
+	}
+
+	// Check if "sensorID" key exists
+	if sensorID, ok := data["sensorID"].(string); ok {
+		// Manipulate the "sensorID"
+		sensorID = splitSensorID(sensorID)
+
```

Comment:
Make sure to run `gofmt`. Most IDEs can be configured to do this every time you save the file.